monitor.h

```diff
@@ -1,14 +1,32 @@
 #pragma once
+#include <vector>
+#include <string>
+
+enum class Unit {
+    Fahrenheit,
+    Celsius
+};
 
 struct Vitals {
     float temperature;
     float pulseRate;
     float spo2;
+    Unit temperatureUnit;  // Added to differentiate the temperature units
 };
 
-struct VitalCheck {
-    bool (*isCritical)(const Vitals&);
-    const char* message;
+class VitalCheck {

```

Comment:
It's a good abstract class. Only one suggestion - give a thought to single responsibility. Does it have one reason for change? This question cannot be answered in the code, we need to think in the domain - from the user's perspective.

- The classification of a vital reading is one aspect that can change - limits can vary based on age etc.
- The summarization message can change: a combination of high temperature, low SPO2 and high pulse rate can point to a lung infection.

Putting the speed of development in focus: With the current structure, will two different developers be able to work on these aspects independently? If not, what kind of split will enable them to do that?